src/script/components/Modals/PrimaryModal/PrimaryModal.tsx

```diff
@@ -165,7 +173,7 @@ export const PrimaryModalComponent: FC = () => {
                       id="clear-data-checkbox"
                       checked={optionChecked}
                       data-uie-name="modal-option-checkbox"
-                      onChange={event => updateOptionChecked(event.target.checked)}
+                      onChange={onOptionChange}
```

Comment:
hmmmm this seems very suscpicious. Do we really want to close the modal when a checkbox is clicked? That's sounds like a weird UX